### PR TITLE
API review: Obsolete ForXxx methods in favor of shorter names

### LINKED ISSUE #16686: Rename ForProviderXxx... methods
Per API review, here are the `ForProviderXxx...` methods so we can decide which, if any, to rename:

### Cosmos

Extends class | Current | Proposal
------------ | ------------- | -------
ModelBuilder | ForCosmosHasDefaultContainer | HasDefaultContainer
EntityTypeBuilder | ForCosmosToContainer | ToContainer
EntityTypeBuilder | ForCosmosHasPartitionKey | HasPartitionKey
PropertyBuilder | ForCosmosToProperty | ToJsonProperty

### SQL Server

Extends class | Current | Proposal
------------ | ------------- | -------
ModelBuilder/PropertyBuilder | ForSqlServerUseSequenceHiLo | UseHiLo
ModelBuilder/PropertyBuilder | ForSqlServerHasHiLoSequence | HasHiLoSequence
ModelBuilder/PropertyBuilder | ForSqlServerUseIdentityColumns | UseIdentityColumns / UseIdentityColumn
ModelBuilder/PropertyBuilder | ForSqlServerHasIdentitySeed | HasIdentityColumnSeed
ModelBuilder/PropertyBuilder | ForSqlServerHasIdentityIncrement | HasIdentityColumnIncrement
ModelBuilder/PropertyBuilder | ForSqlServerHasValueGenerationStrategy | HasValueGenerationStrategy
EntityTypeBuilder | ForSqlServerIsMemoryOptimized | IsMemoryOptimized
IndexBuilder | ForSqlServerIsClustered | IsClustered
IndexBuilder | ForSqlServerInclude |  IncludeColumns
IndexBuilder | ForSqlServerIsCreatedOnline | IsCreatedOnline

### SQLite

Extends class | Current | Proposal
------------ | ------------- | -------
PropertyBuilder | ForSqliteHasSrid | HasSrid
PropertyBuilder | ForSqliteHasDimension | HasSpatialDimension

### In-memory

None

### Npgsql

Extends class | Current | Proposal
------------ | ------------- | -------
ModelBuilder | ForNpgsqlHasEnum | 
ModelBuilder | ForNpgsqlHasMethod | 
ModelBuilder | ForNpgsqlHasRange | 
ModelBuilder | ForNpgsqlUseSerialColumns | 
ModelBuilder | ForNpgsqlUseIdentityAlwaysColumns | 
ModelBuilder | ForNpgsqlUseIdentityByDefaultColumns | 
ModelBuilder | ForNpgsqlUseIdentityColumns | 
ModelBuilder | ForNpgsqlUseTablespace | 
ModelBuilder/PropertyBuilder | ForNpgsqlUseSequenceHiLo | 
EntityTypeBuilder | ForNpgsqlUseXminAsConcurrencyToken | 
EntityTypeBuilder | ForNpgsqlHasIndex | 
EntityTypeBuilder | ForNpgsqlIsUnlogged | 
EntityTypeBuilder | ForNpgsqlSetStorageParameter | 
EntityTypeBuilder/PropertyBuilder | ForNpgsqlHasComment | 
IndexBuilder | ForNpgsqlInclude | 
IndexBuilder | ForNpgsqlHasMethod | 
IndexBuilder | ForNpgsqlHasOperators | 
IndexBuilder | ForNpgsqlHasCollation | 
IndexBuilder | ForNpgsqlHasSortOrder | 
IndexBuilder | ForNpgsqlHasNullSortOrder |

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #16686

### CODE REVIEW

**Comment by ajcvickers** [comment]
@smitpatel @roji Ping for review.

File: src/EFCore.Cosmos/Extensions/CosmosEntityTypeBuilderExtensions.cs [hunk 1]
```diff
@@ -97,7 +97,7 @@ public static class CosmosEntityTypeBuilderExtensions
         /// <param name="entityTypeBuilder"> The builder for the entity type being configured. </param>
         /// <param name="name"> The name of the parent property. </param>
         /// <returns> The same builder instance so that multiple calls can be chained. </returns>
-        public static OwnedNavigationBuilder ForCosmosToProperty(
+        public static OwnedNavigationBuilder ToJsonProperty(
```
Comment (roji) [inline src/EFCore.Cosmos/Extensions/CosmosEntityTypeBuilderExtensions.cs]: Nice solution.

**Comment by roji** [comment]
Now I'll go do the same for EFCore.PG...

**Comment by roji** [comment]
Question: is it intentional to leave the lower-level metadata API for the provider prefix? We now have IsClustered in SqlServerIndexBuilderExtensions, calling into SetSqlServerIsClustered in SqlServerIndexExtensions.

**Comment by ajcvickers** [comment]
@roji I left them because we didn't decide to change them and there may be different considerations. Also, I care less about the lower-level ones. However, we could create an issue to discuss with @AndriySvyryd when he is back.

**Comment by roji** [comment]
@ajcvickers opened #16773 to discuss.

**Pull request merged**